pkg/virt-handler/vm.go

```diff
@@ -1235,9 +1321,13 @@ func (d *VirtualMachineController) execute(key string) error {
 			if uid != "" {
 				log.Log.Object(vmi).V(3).Infof("watchdog file provided %s as UID", uid)
 				vmi.UID = types.UID(uid)
+			} else {
+				// If UID is still not found, cleanup would already have been done
+				return nil
```

Comment:
I'm hesitant to change this here. There could be unintentional side effects. It's possible this is the right thing, but i'm not sure it's worth tackling how to handle VMI's with no UID in this PR. 

in general you shouldn't have to worry about this. If the UID doesn't exist, it's because the VMI has already been deleted and cleaned up. The only code paths at this point involve clean up.  Just make sure the cleanup logic is resilient to UID possibly not existing. 

basically, no UID means that part of the cleanup is a no-op. Take ```err = d.containerDiskMounter.Unmount(vmi)
``` for example, that does nothing if the UID is empty.

Reply:
uid check in Cleanup also look okay to me
Changed it now